linera-chain/src/certificate.rs

```diff
@@ -84,7 +110,42 @@ impl<'de> Deserialize<'de> for ValidatedBlockCertificate {
     where
         D: Deserializer<'de>,
     {
-        Certificate::deserialize(deserializer).map(ValidatedBlockCertificate::from)
+        #[derive(Deserialize)]
+        #[serde(rename = "ValidatedBlockCertificate")]
+        struct Inner {
+            block: ValidatedBlock,
+            round: Round,
+            signatures: Vec<(ValidatorName, Signature)>,
+        }
+        let inner = Inner::deserialize(deserializer)?;
+        let validated_hashed: HashedCertificateValue = inner.block.clone().into();
+        Ok(Self {
+            value: Hashed::unchecked_new(inner.block, validated_hashed.hash()),
+            round: inner.round,
+            signatures: inner.signatures,
+        })
+    }
+}
+
+impl<'de> Deserialize<'de> for TimeoutCertificate {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(rename = "TimeoutCertificate")]
+        struct Inner {
+            timeout: Timeout,
```

Comment:
Can we use `value` here? (and everywhere)

This code could be factorized by the way, but I'm happy to do it later.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2845/commits/fe66e51da6e3a44775b489bde8b7b0d138a13816.